doc/mainpage.dox

```diff
@@ -0,0 +1,66 @@
+// -*- lsst-c++ -*-
+
+namespace lsst { namespace meas { namespace extensions { namespace ngmix {
+/**
+@mainpage lsst::meas::extensions::ngmix;  plugins using NGMIX fitters
```

Comment:
Do we need to block-capitalize "NGMIX"? Upstream seems to refer to it in all lower case, as does the README added in this ticket. It'd be nice to be consistent.